### ISSUE TITLE
REQ: Setting to disable "Output Inspector" auto-show/focus tab in VSCode

### ISSUE THREAD

**Issue by andrewconnell** [wallabyjs/public#2684]
### Issue description or question

Ability to disable **Output Inspector** auto-show window when selecting code in test.

The automatic open & focus of the Output Inspector tab can be quite annoying. While useful, I'd like to open it on demand, not have it automatically hijack focus.

Scenario:
- when creating lots of tests, I stub tests out and set each with a failing assertion, such as adding the single line `expect.assertion(1);` for jest tests
- then, I go back through each test and complete the test to switch from failing to passing
- but when I do this, when I select the dummy placeholder line I added in the first step, the **Output Inspector** window opens, which I then have to close immediately with <kbd>SHIFT</kbd>+<kbd>ESC</kbd>

I can't just go back to my test suite because the line of code is still selected so it automatically takes focus back to the **Output Inspector** window.

settings have an option to set when this window automatically shows, but there's no option to say "never".

### Wallaby diagnostics report

n/a

**Comment by ArtemGovorov** [comment]
Thanks for submitting the request. We'll be working on adding the setting shortly.

One very important thing we need to clarify:

> so it automatically takes focus back to the Output Inspector window.

do you mean mental/attention focus, or your active cursor moves to the Output Inspector window? If it is the latter, can you please share all of your VS Code settings (`Preferences: Open Settings (JSON)` command in VS Code)?

**Comment by smcenlly** [comment]
We have added a new "Never" setting to never display the Output Inspector automatically.

Will wait for your answer to the [question above](https://github.com/wallabyjs/public/issues/2684#issuecomment-818311780) before we close this issue.

> One very important thing we need to clarify:
>
> > so it automatically takes focus back to the Output Inspector window.
>
> do you mean mental/attention focus, or your active cursor moves to the Output Inspector window? If it is the latter, can you please share all of your VS Code settings (Preferences: Open Settings (JSON) command in VS Code)?

**Comment by andrewconnell** [comment]
I'm not sure what was different the last few days, but the behavior is now entirely different than it was when I submitted this. IOW, I can't repro my issue. I'll skip sharing my settings as I can't repro so seems non-actionable.

Let me explain further:
- I had one instance of VS Code, with two files open in different tabs: my SUT (tab1) & tests file (tab2)
- when writing my tests, my style is to write out all the conditions I should test for but add a single line to fail the test (since jest doesn't provide an *inconclusive* result option): `expect.assertions(1);`. that would fail the test as it's the only line in the test and thus, creating my "work task list"
- as I'd work through the tests, using the keyboard, I'd select the entire LOC above
  - this triggered Wallaby to open the **Output Inspector** window in tab2... but because my spec is loaded in tab2, it would hijack focus
  - but what was really annoying that initiated me to submit this issue, was that when I tried to toggle back to my spec file, because the LOC was still selected, Wallaby would auto-focus back to the **Object Inspector** tab
- thus, I was suck in an loop until I forced closed the **Object Inspector** window... doing so with the mouse or closing the tab didn't help as it would just fire up again... but doing so with the <kbd>SHIFT</kbd>+<kbd>ESC</kbd> combo worked

As I said, this isn't happening now & I don't understand why as I didn't change anything other than restart VS Code overnight.

**Comment by smcenlly** [comment]
> I'm not sure what was different the last few days, but the behavior is now entirely different than it was when I submitted this.

We had inadvertently broken the show error on selection feature unless you had the setting `wallaby.outputInspectorFeature.automaticDisplay` set to `Errors Only`.

Please update to Wallaby for VS Code v1.0.284 and the behavior should be the same as when you submitted the issue. Of course, you may now select `Never` as your `Output Inspector` automatic display setting to fix the issue.

Having said that, we tried to reproduce your problem and couldn't. Can you detail the exact steps that you followed? What keyboard shortcuts are you using and what commands are they bound to? We tried `View: Focus Next Editor Group` and `View: Focus Previous Editor Group`.

**Comment by andrewconnell** [comment]
I’m not able to repro or explain the steps I did that triggered this behavior unfortunately... so I’ll go ahead and close this as I can’t say it’s even an issue anymore.

To your ask about keyboard shortcuts & what they’re bound to... I don’t have many extensions installed, in fact I uninstalled everything and only added a few back after having perf issues with VSCode. Nothing exotic & nothing customized.
